sdk/identity/azure-identity/README.md

```diff
@@ -0,0 +1,186 @@
+
+# Examples
+
+## Authenticating with `DefaultAzureCredential`
+This example demonstrates authenticating the `SecretClient` from the [azure-keyvault-secrets][secrets_client_library] client library using the `DefaultAzureCredential`.
+```java
+// The default credential first checks environment variables for configuration as described above.
+// If environment configuration is incomplete, it will try managed identity.
+import com.azure.identity.credential.DefaultAzureCredential;
+import com.azure.security.keyvault.secrets.SecretClient;
+
+DefaultAzureCredential defaultCredential = new DefaultAzureCredential();
+
+// Azure SDK client builders accept the credential as a parameter
+
+SecretClient client = SecretClient.builder()
+    .endpoint("https://{YOUR_VAULT_NAME}.vault.azure.net")
+    .credential(credential)
+    .build();
+```
+When executing this in a development machine you need to first [configure the environment](#environment-variables) setting the variables `AZURE_CLIENT_ID`, `AZURE_TENANT_ID` and `AZURE_CLIENT_SECRET` to the appropriate values for your service principal.
+
+## Authenticating a service principal with a client secret
+This example demonstrates authenticating the `KeyClient` from the [azure-keyvault-keys][keys_client_library] client library using the `ClientSecretCredential`.
+```java
+// using a client secret
+import com.azure.identity.credential.ClientSecretCredential;
+import com.azure.security.keyvault.keys.KeyClient;
+
+// authenticate with client secret,
+ClientSecretCredential clientSecretCredential = new ClientSecretCredential()
+	    .clientId("<YOUR_CLIENT_ID>")
+	    .clientSecret("<YOUR_CLIENT_SECRET>")
+	    .tenantId("<YOUR_TENANT_ID>");
+
+KeyClient client = KeyClient.builder()
+    .endpoint("https://{YOUR_VAULT_NAME}.vault.azure.net")
+    .credential(clientSecretCredential)
+    .build();
+```
+
+## Chaining credentials:
+The `ChainedTokenCredential` class provides the ability to link together multiple credential instances to be tried sequentially when authenticating. The following example demonstrates creating a credential which will attempt to authenticate using managed identity, and fall back to certificate authentication if a managed identity is unavailable in the current environment. This example authenticates an `EventHubClient` from the [azure-eventhubs][eventhubs_client_library] client library using the `ChainedTokenCredential`.
+
+```java
+import com.azure.identity.credential.ClientSecretCredential;
+import com.azure.security.keyvault.secrets.SecretClient;
+
+ClientSecretcredential firstServicePrincipal = new ClientSecretCredential()
+	    .clientId("<YOUR_CLIENT_ID>")
+	    .clientSecret("<YOUR_CLIENT_SECRET>")
+	    .tenantId("<YOUR_TENANT_ID>");
+
+ClientSecretcredential secondServicePrincipal = new ClientSecretCredential()
+	    .clientId("<YOUR_CLIENT_ID>")
+	    .clientSecret("<YOUR_CLIENT_SECRET>")
+	    .tenantId("<YOUR_TENANT_ID>");
```

Comment:
I would use two different authentication types here, such as ManagedIdentityCredential followed by ClientSecretCredential, I think it demonstrates a more common use case than two different ClientSecretCredential instances.

Reply:
Updated.